org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/rename/RenameProcessor.java

```diff
@@ -122,11 +130,20 @@ protected SearchPattern createOccurrenceSearchPattern(IJavaElement[] elements) t
 	protected void convert(WorkspaceEdit root, ICompilationUnit unit, TextEdit edits) {
 		TextEditConverter converter = new TextEditConverter(unit, edits);
 		String uri = JDTUtils.toURI(unit);
-		Map<String, List<org.eclipse.lsp4j.TextEdit>> changes = root.getChanges();
-		if (changes.containsKey(uri)) {
-			changes.get(uri).addAll(converter.convert());
+		if (preferenceManager.getClientPreferences().isWorkspaceEditResourceChangesSupported()) {
+			List<Either<ResourceChange, TextDocumentEdit>> changes = root.getResourceChanges();
+			if (changes == null) {

```

Comment:
Just wondering how the ordering constraints on the resource changes vs. text changes are.